### PR TITLE
Add a deploy-managed target similar to deploy-kind

### PR DESCRIPTION
**Pull request by priyaselvaganesan**

### CODE REVIEW

**Review by m25n — APPROVED** [review]
LGTM!

File: Makefile [hunk 1]
```diff
@@ -203,7 +203,13 @@ GOOS= GOARCH= GOBIN=$(PROJECT_DIR)/bin go install $(2) ;\
 }
 endef
 
-deploy-kind: copy-kind-patches docker-build manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+deploy-managed: docker-xplatform-build ## Deploy controller to the K8s cluster specified in ~/.kube/config.
```
Comment (m25n) [inline Makefile:206]: On second thought, we should probably add `manifests` and `kustomize` as build dependencies.

**Review by m25n — CHANGES_REQUESTED** [review]
I noticed that we should probably have some other build dependencies.

**Review by m25n — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (priyaselvaganesan) [reply Makefile]: Thank you for the catch!!

**Pull request merged**